units/recalculate_change_points.go

```diff
@@ -42,200 +42,88 @@ func makeChangePointsJob() *recalculateChangePointsJob {
 	return j
 }
 
-func NewRecalculateChangePointsJob(metricGrouping MetricGrouping) amboy.Job {
+func NewRecalculateChangePointsJob(timeSeriesId model.TimeSeriesId) amboy.Job {
 	j := makeChangePointsJob()
 	// Every ten minutes at most
 	timestamp := util.RoundPartOfHour(10)
-	j.SetID(fmt.Sprintf("%s.%s.%s.%s.%s.%s", j.JobType.Name, metricGrouping.Id.Project, metricGrouping.Id.Variant, metricGrouping.Id.Task, metricGrouping.Id.Test, timestamp))
-	j.metricGrouping = metricGrouping
+	j.SetID(fmt.Sprintf("%s.%s.%s.%s.%s.%s.%s", j.JobType.Name, timeSeriesId.Id.Project, timeSeriesId.Id.Variant, timeSeriesId.Id.Task, timeSeriesId.Id.Test, timeSeriesId.Id.Measurement, timestamp))
+	j.TimeSeriesId = timeSeriesId
 	return j
 }
 
-type MetricGrouping struct {
-	Id struct {
-		Project string `bson:"project"`
-		Variant string `bson:"variant"`
-		Task    string `bson:"task"`
-		Test    string `bson:"test"`
-	} `bson:"_id"`
-}
-
-func makeSeriesFilter(grouping MetricGrouping) bson.M {
-	return bson.M{
-		"info.project":   grouping.Id.Project,
-		"info.variant":   grouping.Id.Variant,
-		"info.task_name": grouping.Id.Task,
-		"info.test_name": grouping.Id.Test,
+func makeMessage(msg string, id model.TimeSeriesId) message.Fields {
```

Comment:
make this a method on your job?

Reply:
Why? Won't we just be adding characters for the same functionality?

Reply:
It's clearer, and takes advantage of an existing namespace (in a package which may eventually have a lot of "messages" to make". 

I also think it's good ot do things like put the j.ID() in to the messages, so you can check which execution writes a given message.